### PR TITLE
Autoware plugin mininal version

### PR DESCRIPTION
**Pull request by qswawrq**
# PR Details
This is the initial implementation of Autoware plugin.
## Description
This plugin takes in a list of waypoints from Autoware and convert them into a list of evenly spaced trajectory points.
## Related Issue
N/A
## Motivation and Context
Adding new feature for Beta release
## How Has This Been Tested?
All unit tests passed.
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: autoware_plugin/src/autoware_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        starting_point.x = pose_msg_->pose.position.x;
+        starting_point.y = pose_msg_->pose.position.y;
+        uneven_traj.push_back(starting_point);
+    }
+    for(int i = 0; i < waypoints.size(); ++i)
+    {
+        double previous_wp_v = current_speed_;
+        double previous_wp_x = pose_msg_->pose.position.x;
+        double previous_wp_y = pose_msg_->pose.position.y;
+        double previous_wp_t = 0.0;
```
Comment (Supraluminal) [inline autoware_plugin/src/autoware_plugin.cpp]: Doesn't matter a lot here, but normally you can make things more efficient by minimizing the amount of memory allocation (stack or heap) done inside of a loop, so if you move these outside of the for loop they only have to be allocated once.

File: autoware_plugin/include/autoware_plugin.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#include <vector>
+#include <cav_msgs/TrajectoryPlan.h>
+#include <cav_msgs/TrajectoryPlanPoint.h>
+#include <autoware_msgs/Lane.h>
+#include <boost/shared_ptr.hpp>
+#include <carma_utils/CARMAUtils.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+
+class AutowarePlugin
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/include/autoware_plugin.h]: Yes. It should be in the its own namespace matching to its project name..

File: autoware_plugin/src/autoware_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include "autoware_plugin.h"
+
+AutowarePlugin::AutowarePlugin() :
+                current_speed_(0.0),
+                trajectory_time_length_(6.0),
+                trajectory_point_spacing_(0.1) {}
+
+AutowarePlugin::~AutowarePlugin() {}
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/src/autoware_plugin.cpp]: Got u.

File: autoware_plugin/src/autoware_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        {
+            break;
+        }
+    }
+    return sublist;
+}
+
+std::vector<cav_msgs::TrajectoryPlanPoint> AutowarePlugin::even_trajectory(std::vector<cav_msgs::TrajectoryPlanPoint> trajectory, double time_spacing)
+{
+    std::cout << "Function is called!" << std::endl;
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/src/autoware_plugin.cpp]: Nope. Deleted.

File: autoware_plugin/src/autoware_plugin.cpp [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    return res;
+}
+
+std::vector<cav_msgs::TrajectoryPlanPoint> AutowarePlugin::post_process_traj_points(std::vector<cav_msgs::TrajectoryPlanPoint> trajectory)
+{
+    for(int i = 0; i < trajectory.size(); ++i)
+    {
+        trajectory[i].controller_plugin_name = "pure_pursuit";
+        trajectory[i].planner_plugin_name = "autoware";
+        trajectory[i].target_time += ros::Time::now().toNSec();
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/src/autoware_plugin.cpp]: Good catch!

File: autoware_plugin/src/autoware_plugin.cpp (hunk 1 above)
Comment (qswawrq) [reply autoware_plugin/src/autoware_plugin.cpp]: Sure. It makes sense to me.

File: autoware_plugin/src/autoware_plugin.cpp [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+std::vector<cav_msgs::TrajectoryPlanPoint> AutowarePlugin::even_trajectory(std::vector<cav_msgs::TrajectoryPlanPoint> trajectory, double time_spacing)
+{
+    std::cout << "Function is called!" << std::endl;
+    std::vector<cav_msgs::TrajectoryPlanPoint> res;
+    res.push_back(trajectory[0]);
+    double tp_timestamp = 0.0 + time_spacing * 1e9;
+    int next_traj_point_index = 1;
+    while(true)
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/src/autoware_plugin.cpp]: Yes. I should do that.

File: autoware_plugin/include/autoware_plugin.h [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+{
+
+public:
+
+    AutowarePlugin();
+    virtual ~AutowarePlugin();
```
[the remark this answers is not in the document]
Comment (qswawrq) [reply autoware_plugin/include/autoware_plugin.h]: Defining a constructor is only for using the initializer lists here. I am not sure if an explicit copy constructor and a copy assignment constructor is necessary here?

**Pull request merged**